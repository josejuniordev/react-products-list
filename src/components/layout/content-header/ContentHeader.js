import React from 'react';
import PropTypes from 'prop-types';
import './ContentHeader.scss';

function ContentHeader(
  {
    title,
    titleComplement,
    description,
  }
) {
  return (
    <header className='content-header'>
      <h1 className='content-header__title'>
        {title}
        {
          titleComplement && (
            <span className='content-header__title-complement'> - {titleComplement}</span>
          )
        }
      </h1>
      <p className='content-header__description'>{description}</p>
    </header>
  )
}

ContentHeader.propTypes = {
  title: PropTypes.string,
  titleComplement: PropTypes.string,
  description: PropTypes.string,
};

ContentHeader.defaultProps = {
  title: '',
  titleComplement: '',
  description: '',
};

export default ContentHeader;
