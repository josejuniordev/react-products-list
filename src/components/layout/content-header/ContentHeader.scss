@import "../../../styles/colors";
@import "../../../styles/bem";

.content-header {
  color: inherit;
  border-bottom: 1px solid $separator-color;
  padding-bottom: 50px;
  margin-bottom: 50px;

  @include element(title) {
    font-size: 20px;
    font-weight: bolder;
    margin-right: 15px;
  }

  @include element(title-complement) {
    font-size: inherit;
    font-weight: lighter;
  }

  @include element(description) {
    color: $secondary-text-color;
  }

  @include element(container) {
    display: flex;
    align-items: flex-start;
  }

  @include element(start-enhancer) {
    margin-right: 15px;
  }

  @include element(end-enhancer) {
    flex: 2;
    text-align: right;
  }
}
