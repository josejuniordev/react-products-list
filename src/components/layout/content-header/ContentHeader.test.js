import { shallow, render } from 'enzyme';
import React from 'react';
import ContentHeader from './ContentHeader';

describe('<ContentHeader />', () => {
  const title = 'Empresa XPTO';
  const titleComplement = 'Lista de produtos';
  const description = 'Descrição do header';
  const renderedWrapper = render(
    <ContentHeader title={title} titleComplement={titleComplement} description={description} />,
  );

  it('should render correctly', () => {
    expect(renderedWrapper).toMatchSnapshot();
  });

  const wrapper = shallow(
    <ContentHeader title={title} titleComplement={titleComplement} description={description} />,
  );

  describe('should render a title tag correctly', () => {
    describe('with the complementary text', () => {
      const foundTitle = wrapper.find('h1');

      it('should have a h1 html tag', () => {
        expect(foundTitle.length).toEqual(1);
      });

      it('should have the correct text merge in the title', () => {
        expect(foundTitle.text()).toEqual(`${title} - ${titleComplement}`);
      });
    });

    describe('without the complementary text', () => {
      const wrapperWithoutComplement = shallow(
        <ContentHeader title={title} description={description} />,
      );
      const foundTitle = wrapperWithoutComplement.find('h1');

      it('should have a h1 html tag', () => {
        expect(foundTitle.length).toEqual(1);
      });

      it('should have the correct text merge in the title', () => {
        expect(foundTitle.text()).toEqual(title);
      });
    });
  });

  describe('should render a title complement tag correctly', () => {
    const foundTitleComplement = wrapper.find('h1 span');

    it('should have a span html tag', () => {
      expect(foundTitleComplement.length).toEqual(1);
    });

    it('should have the right title complement text', () => {
      expect(foundTitleComplement.text()).toEqual(` - ${titleComplement}`);
    });
  });

  describe('should render header description tag correctly', () => {
    const foundHeaderDescription = wrapper.find('p');

    it('should have a p html tag', () => {
      expect(foundHeaderDescription.length).toEqual(1);
    });

    it('should have the right title text', () => {
      expect(foundHeaderDescription.text()).toEqual(description);
    });
  });
});
