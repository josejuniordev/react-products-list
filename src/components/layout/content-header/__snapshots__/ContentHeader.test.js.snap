// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentHeader /> should render correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "content-header",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "content-header__title",
        },
        "children": Array [
          Object {
            "data": "Empresa XPTO",
            "next": Object {
              "attribs": Object {
                "class": "content-header__title-complement",
              },
              "children": Array [
                Object {
                  "data": " - Lista de produtos",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
          Object {
            "attribs": Object {
              "class": "content-header__title-complement",
            },
            "children": Array [
              Object {
                "data": " - Lista de produtos",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "data": "Empresa XPTO",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "h1",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "content-header__description",
          },
          "children": Array [
            Object {
              "data": "Descrição do header",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "content-header__description",
        },
        "children": Array [
          Object {
            "data": "Descrição do header",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "p",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "content-header__title",
          },
          "children": Array [
            Object {
              "data": "Empresa XPTO",
              "next": Object {
                "attribs": Object {
                  "class": "content-header__title-complement",
                },
                "children": Array [
                  Object {
                    "data": " - Lista de produtos",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
            Object {
              "attribs": Object {
                "class": "content-header__title-complement",
              },
              "children": Array [
                Object {
                  "data": " - Lista de produtos",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "data": "Empresa XPTO",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "h1",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "header",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
